pkg/dataplane/client.go

```diff
@@ -35,6 +63,63 @@ func NewClient(aud, cloud string, cred azcore.TokenCredential) (*ManagedIdentity
 	return &ManagedIdentityClient{swaggerClient: swaggerClient}, nil
 }
 
+func (c *ManagedIdentityClient) GetUserAssignedMSI(ctx context.Context, request UserAssignedMSIRequest) (*CredentialsObject, error) {
+	validate := validator.New(validator.WithRequiredStructEnabled())
+	if err := validate.Struct(request); err != nil {
+		return nil, fmt.Errorf("%w: %w", errInvalidRequest, err)
+	}
+
+	ctx = context.WithValue(ctx, identityURLKey, request.IdentityURL)
+	credRequestDef := swagger.CredRequestDefinition{
+		IdentityIDs: []*string{&request.ResourceID},
+	}
+
+	creds, err := c.swaggerClient.Getcreds(ctx, credRequestDef, nil)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", errGetCreds, err)
+	}
+
+	//
+	// GetCreds can return multiple identities. We expect one. Return if we don't find one.
+	//
+	// If mulitple identities are found, we return because we don't want to accidentally process
+	// an identitiy that wasn't requested.
+	//
+	if len(creds.ExplicitIdentities) != 1 {
+		return nil, fmt.Errorf("%w, found %d", errNotOneMSI, len(creds.ExplicitIdentities))
```

Comment:
```suggestion
		return nil, fmt.Errorf("%w, found %d identifies instead", errNotOneMSI, len(creds.ExplicitIdentities))
```
suggestion to make the log easier to read